src/Controller/ResearchPlanController.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Controller;
+
+use App\Repository\ResearchRequestRepository;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Entity;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class ResearchPlanController extends AbstractController
+{
+    #[Route('/research-plan/{id}', name: 'app_research_plan', methods: ['GET', 'POST'])]
+    #[Entity('researchRequest', options: ['id' => 'researchRequestId'])]
+    public function index(int $id, ResearchRequestRepository $researchRequestRepo): Response
```

Comment:
Why don't you directly ask for a ResearchRequest in the param of your action, instead of an id and a repo ?

Reply:
it's actually easier. thank you